Module code adapted for PHPStan (#1)
TavoNiievez/module-symfony: CONTRIBUTING.md

# How to Contribute

First of all: Contributions are very welcome!

> For *larger* changes, we recommend to use the test project for this module: [Codeception/symfony-module-test](https://github.com/Codeception/symfony-module-tests). It is a minimal (but complete) Symfony project, ready to run tests. The code inside `vendor/codeception/module-symfony` is the same as...

Author: OK, but how would i know how 'long' my changes are to choose to use the test project vs the Github editor?

You should test your change in the test project if you are going to modify the internal code of a function or you will add a totally new one, but if you will only add comments, or change the name of some variable it may not be so necessary.

Reviewer: See today's Commit :-)

# How to Contribute

First of all: Contributions are very welcome!

Does your change require a test?

## Yes, My Change Requires a Test

(i.e. you add or modify functionality)

1. Install the test project for this module: [Codeception/symfony-module-test](https://github.com/Codeception/symfony-module-tests). It is a minimal (but complete) Symfony project, ready to run tests.

2. Edit the modules source code in `vendor/codeception/module-symfony/src/Codeception/Module/Symfony.php` of this project.

3. Create a test with the same name as your new function in `tests/Functional/SymfonyModuleCest.php`, following alphabetical order.

> 

Author: Here would come the explanation of `vendor/bin/codecept clean` and `vendor/bin/codecept build`

4. Run the tests with `vendor/bin/codecept run Functional`.

> 5. ???

Author: anything else that needs clarification?

# How to Contribute

First of all: Contributions are very welcome!

Does your change require a test?

## Yes, My Change Requires a Test

So you're going to add or modify functionality? Hey, you rock!

You can use our prepared [Codeception/symfony-module-test](https://github.com/Codeception/symfony-module-tests). It is a minimal (but complete) Symfony project, ready to run tests.

1. On https://github.com/Codeception/symfony-module-tests, click on the "Fork" button. Then, in your terminal, do:
  ```bash
  git clone https://github.com/YourUserName/symfony-module-tests.git
  cd symfony-module-tests
  git checkout -b new_feature

>   composer update

Author: we shouldn't include this step, it depends on the php version the user is running and is already explained in the test project readme.md
```suggestion
```

Author: I'm going to merge this without the `composer update`

   ```bash
   git clone https://github.com/YourUserName/module-symfony.git
   cd module-symfony
   git checkout -b new_feature
   ```

4. Copy your changed code parts from the test project's `Symfony.php` to this fork's `src/Codeception/Module/Symfony.php`

5. Commit:
  ```bash
  git add --all
  git commit --message="Briefly explain what your change is about"
  git push --set-upstream origin new_feature
  ```

6. In the CLI output, click on the link to https://github.com/YourUserName/module-symfony/pull/new/new_feature to create a Pull Request through GitHub.com.

Now wait for feedback on your Pull Request.  
If all is fine, then ...

### ... Write the Test

7. In the test project (`symfony-module-tests`), create a test with the same name as your new function in `tests/Functional/SymfonyModuleCest.php`, following alphabetical order.
  Hint: Run this to rebuild Codeception's "Actor" classes (see [Console Commands](https://codeception.com/docs/reference/Commands#Build)) to get auto-completion in your IDE:
  ```bash
  vendor/bin/codecept clean
  vendor/bin/codecept build
  ```

8. Run the tests with `vendor/bin/codecept run Functional`

9. Commit:
  ```bash

>   git add -A
>   git commit -m "Add a link to the module's Pull Request you created above"

Author: ```suggestion
  git add --all
  git commit --message="Add a link to the module's Pull Request you created above"
```

# How to Contribute

First of all: Contributions are very welcome!

Does your change require a test?

## Yes, My Change Requires a Test

So you're going to add or modify functionality? Hey, you rock!

1. Install the test project for this module: [Codeception/symfony-module-test](https://github.com/Codeception/symfony-module-tests). It is a minimal (but complete) Symfony project, ready to run tests.

2. Edit the module's source code in `vendor/codeception/module-symfony/src/Codeception/Module/Symfony.php` of this project.

3. On [GitHub.com](https://github.com/Codeception/module-symfony), click on the "Fork" button. Then, in your terminal, go to another directory, then:
   ```bash
   git clone https://github.com/YourUserName/module-symfony.git
   cd module-symfony
   git checkout -b new_feature
   ```

4. Copy your changed code parts from the test project's `Symfony.php` to this fork's `src/Codeception/Module/Symfony.php`

5. Commit:
  ```bash

>   git add -A
>   git commit -m "Briefly explain what your change is about"

Author: ```suggestion
  git commit --all --message="Briefly explain what your change is about"
```

Author: if the` git add` is mandatory then just replace `-a` with `--all`

2. Edit the modules source code in `vendor/codeception/module-symfony/src/Codeception/Module/Symfony.php` of this project.

3. Create a test with the same name as your new function in `tests/Functional/SymfonyModuleCest.php`, following alphabetical order.

>   Hint: Run this to rebuild Codeception's "Actor" classes (see [Console Commands](https://codeception.com/docs/reference/Commands#Build)) to get auto-completion in your IDE:

Author: Cool!

Reviewer: :-)
